app/lib/katello/validators/environment_docker_repositories_validator.rb

```diff
@@ -20,19 +20,21 @@ def validate(environment)
 
       def self.validate_repositories(registry_name_pattern, repositories)
         error_messages = []
-        names = []
+        name_to_repos = {}
         repositories.each do |repository|
           name = Katello::Repository.safe_render_container_name(repository, registry_name_pattern)
-
           unless ContainerImageNameValidator.validate_name(name)
             error_messages << N_("Registry name pattern results in invalid container image name of member repository '%{name}'") % {name: repository.name}
             return error_messages
           end
-          names << name
+          name_to_repos[name] ||= []
+          name_to_repos[name] << repository
         end
 
-        if names.length != names.uniq.length
-          error_messages << N_("Registry name pattern results in duplicate container image names")
+        duplicate_repos = name_to_repos.select { |_name, repos| repos.count > 1 }.values.flatten
+        if duplicate_repos.any?
+          repo_names = duplicate_repos.map(&:name).join(', ')
+          error_messages << N_("Registry name pattern results in duplicate container image names for these repositories: %s") % repo_names
```

Comment:
Can we add "Consider changing the Lifecycle Environment's Registry Name Pattern to something more specific." at the end of this message?

Right now if I try to promote second CV, I get nice, detailed error message. If I try to promote one CV with two repos, I get error that tells me what is wrong (improvement over what we had!), but doesn't point me in direction to how I should fix it.

Reply:
Error should now be:

```Registry name pattern results in duplicate container image names for these repositories: docker, docker2. Consider changing the Lifecycle Environment's Registry Name Pattern to something more specific.```

Reply:
Confirmed. This is the message I see now. Thanks!